solo/models.py

```diff
@@ -1,63 +1,70 @@
+from __future__ import annotations
+
+import sys
+from typing import Any
+
 from django.conf import settings
+from django.core.cache import caches
 from django.db import models
 
-try:
-    from django.core.cache import caches
-    get_cache = lambda cache_name: caches[cache_name]
-except ImportError:
-    from django.core.cache import get_cache
-
 from solo import settings as solo_settings
 
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+
 DEFAULT_SINGLETON_INSTANCE_ID = 1
 
+
 class SingletonModel(models.Model):
     singleton_instance_id = DEFAULT_SINGLETON_INSTANCE_ID
 
     class Meta:
         abstract = True
 
-    def save(self, *args, **kwargs):
+    def save(self, *args: Any, **kwargs: Any) -> None:
         self.pk = self.singleton_instance_id
-        super(SingletonModel, self).save(*args, **kwargs)
+        super().save(*args, **kwargs)
         self.set_to_cache()
 
-    def delete(self, *args, **kwargs):
+    def delete(self, *args: Any, **kwargs: Any) -> tuple[int, dict[str, int]]:
         self.clear_cache()
-        super(SingletonModel, self).delete(*args, **kwargs)
+        return super().delete(*args, **kwargs)
```

Comment:
This is a runtime change, but imo better this way

Reply:
This is not a breaking change though, as previous users simply would not have used a return value at all?

Reply:
Yes, just wanted to make sure you were aware of it :)